scripts/sonic_sku_create.py

```diff
@@ -0,0 +1,457 @@
+#! /usr/bin/env python2
+"""
+usage: sonic_sku_create.py [-h] [-v] [-f FILE] [-m [MINIGRAPH_FILE]] [-b BASE]
+                           [-r] [-c [{new_sku_only,l2_mode_only}]] [-k HWSKU]
+                           [-p] [-vv]
+Create a new SKU
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -v, --version         show program's version number and exit
+  -f FILE, --file FILE  SKU definition from xml file. -f OR -m must be provided when creating a new SKU
+  -m [MINIGRAPH_FILE], --minigraph_file [MINIGRAPH_FILE]
+                        SKU definition from minigraph file. -f OR -m must be provided when creating a new SKU
+  -b BASE, --base BASE  SKU base definition
+  -r, --remove          Remove SKU folder
+  -c [{new_sku_only,l2_mode_only}], --cmd [{new_sku_only,l2_mode_only}]
+                        Choose action to preform (Generate a new SKU, Configure L2 mode, Both
+  -k HWSKU, --hwsku HWSKU
+                        SKU name to be used when creating a new SKU or for  L2 configuration mode
+  -p, --print           Print port_config.ini without creating a new SKU
+  -vv, --verbose        Verbose output
+
+
+"""
+
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import re
+import subprocess
+import swsssdk
+import traceback
+import sys
+import shutil
+
+from tabulate import tabulate
+from lxml import etree as ET
+from lxml.etree import QName
+
+minigraph_ns = "Microsoft.Search.Autopilot.Evolution"
+minigraph_ns1 = "http://schemas.datacontract.org/2004/07/Microsoft.Search.Autopilot.Evolution"
+platform = None
+
+test_envir = None
+try:
+	test_envir = os.environ.get('UTILITIES_UNIT_TESTING')
+except KeyError:
+	print ("Not Runing on unit test envir", file=sys.stderr)
+
+### port_config.ini header
+PORTCONFIG_HEADER = ["# name", "lanes", "alias", "index", "speed"]
+
+
+class SkuCreate(object):
+	"""
+	Process aclstat
+	"""
+
+	def __init__(self):
+
+		self.portconfig_dict = {}
+		self.platform_specific_dict = {"x86_64-mlnx_msn2700-r0":self.msn2700_specific}
+		self.default_lanes_per_port = []
+		self.fpp = []
+		self.fpp_split  =  {}
+		self.num_of_fpp = 0
+		self.num_of_lanes_dict = {"1" : 4, "2" : 2, "4" : 1}
```

Comment:
if this is based on 2700 default, we should not use it. we have already 8 lanes support for different ASICs.

Reply:
There were no assumptions made on number of lanes. Removed the line in the next commit.